### PR TITLE
Improve Astro libraries config handling

### PR DESCRIPTION
**Pull request by bluwy**
## Changes

Use https://github.com/svitejs/vitefu to handle third-party Astro libraries SSR config. This should make most work out of the box.

## Testing

Existing tests should pass. Hope no Windows issue.

## Docs

N/A

### CODE REVIEW

File: packages/astro/src/core/create-vite.ts [hunk 1]
```diff
@@ -58,7 +56,31 @@ export async function createVite(
 	commandConfig: vite.InlineConfig,
 	{ settings, logging, mode }: CreateViteOptions
 ): Promise<vite.InlineConfig> {
-	const thirdPartyAstroPackages = await getAstroPackages(settings);
+	const astroPkgsConfig = await crawlFrameworkPkgs({
```
Comment (matthewp) [inline packages/astro/src/core/create-vite.ts:59]: very nice API!

File: pnpm-lock.yaml [hunk 2]
```diff
@@ -455,6 +455,7 @@ importers:
       unist-util-visit: ^4.1.0
       vfile: ^5.3.2
       vite: ~3.1.3
+      vitefu: ^0.0.1
```
Comment (matthewp) [inline pnpm-lock.yaml:458]: i see that this is your package, what do you think about bumping this to 0.1.0? I'd hate for us to have to upgrade every time you fix a bug there.

Comment (bluwy) [reply pnpm-lock.yaml:458]: Yeah I'm planning to do so quite soon once I get the API ironed out with Dominik from `vite-plugin-svelte`. And making sure it really works.

I can cut 0.1.0 once this CI passes, along with https://github.com/sveltejs/vite-plugin-svelte/pull/478 and an upcoming PR to vite-plugin-solid

Comment (bluwy) [reply pnpm-lock.yaml]: Released and updated to 0.1.0 👍

**Review by matthewp — APPROVED** [review]

**Pull request merged**